File: docs/engineer/dip/colorbar/py/colorbar.py
import numpy as np
import cv2 

# Create a black image
img = np.zeros((80,560,3), np.uint8)
#img.fill(200)

# BGR (not RGB)

# gray
cv2.rectangle(img,   (0,0),  (80,80), (0xC0, 0xC0, 0xC0), -1)
# yellow
cv2.rectangle(img,  (80,0), (160,80), (0x00, 0xFF, 0xFF), -1)
# cyan
cv2.rectangle(img, (160,0), (240,80), (0xFF, 0xFF, 0x00), -1)
# green
cv2.rectangle(img, (240,0), (320,80), (0x00, 0xFF, 0x00), -1)
# magenta
cv2.rectangle(img, (320,0), (400,80), (0xFF, 0x00, 0xFF), -1)
# red
cv2.rectangle(img, (400,0), (480,80), (0x00, 0x00, 0xFF), -1)
# blue
cv2.rectangle(img, (480,0), (560,80), (0xFF, 0x00, 0x00), -1)

cv2.imwrite('colorbar.png',img)

File: docs/engineering/courses/undergraduate/dip/php/colorbar/colorbar.php
<?php
$im = imagecreatetruecolor(560, 80);

$gray    = imagecolorallocate($im, 0xC0, 0xC0, 0xC0);
$yellow  = imagecolorallocate($im, 0xFF, 0xFF, 0x00);
$cyan    = imagecolorallocate($im, 0x00, 0xFF, 0XFF);
$green   = imagecolorallocate($im, 0x00, 0xFF, 0x00);
$magenta = imagecolorallocate($im, 0xFF, 0x00, 0xFF);
$red     = imagecolorallocate($im, 0xFF, 0x00, 0x00);
$blue    = imagecolorallocate($im, 0x00, 0x00, 0xFF);

imagefilledrectangle($im,   0, 0,  80, 80, $gray);
imagefilledrectangle($im,  80, 0, 160, 80, $yellow);
imagefilledrectangle($im, 160, 0, 240, 80, $cyan);
imagefilledrectangle($im, 240, 0, 320, 80, $green);
imagefilledrectangle($im, 320, 0, 400, 80, $magenta);
imagefilledrectangle($im, 400, 0, 480, 80, $red);
imagefilledrectangle($im, 480, 0, 560, 80, $blue);

imagepng($im,'colorbar.png');
imagedestroy($im);
?>